src/main/java/algorithms/curated170/hard/EncodeStringWithShortestLengthDP.java

```diff
@@ -0,0 +1,56 @@
+package algorithms.curated170.hard;
+
+public class EncodeStringWithShortestLengthDP {
+
+    public String encode(String s) {
+        int n = s.length();
+        String[][] codings = new String[n][n];
+
+        for (int e = 0; e < n; e++) {
+
+            initializeEndings(s, codings, e);
+
+            for (int b = e - 1; b - (e - b) + 1 >= 0; --b) {
+                String pattern = s.substring(b + 1, e + 1);
+                int patternLen = e - b;
+
+                int maxRepStartIdx = b - patternLen + 1;
+                for (int repStart = maxRepStartIdx; repStart >= 0
+                        && pattern.equals(s.substring(repStart, repStart + patternLen)); repStart -= patternLen) {
```

Comment:
Thanks, Erdem.
It is remarkably cleaner now.
Let us lighten the burden of for-loop to make it simpler:
```suggestion
    for (int repStart = maxRepStartIdx; repStart >= 0; repStart -= patternLen) {
                   if(!pattern.equals(s.substring(repStart, repStart + patternLen)))
                    break;
```